src/lib.rs

```diff
@@ -503,23 +517,50 @@ impl Contract {
         handle: CommunityHandle,
         #[allow(unused_mut)] mut community: Community,
     ) {
-        let target_community = self
+        let _ = self
             .get_editable_community(&handle)
             .expect("Only community admins and hub moderators can configure communities");
 
         community.validate();
         community.set_default_admin();
 
-        if target_community.handle == community.handle {
-            self.communities.insert(&target_community.handle, &community);
-        } else {
-            require!(
-                self.communities.get(&community.handle).is_none(),
-                "Community handle is already taken"
-            );
-            self.communities.remove(&target_community.handle);
-            self.communities.insert(&community.handle, &community);
-        }
+        require!(community.handle == handle, "Community handle cannot be changed");
+        require!(env::prepaid_gas() >= UPDATE_COMMUNITY_GAS, "Require at least 30 Tgas");
+        self.communities.insert(&handle, &community);
+        ext_social_db::ext(SOCIAL_DB.parse().unwrap()).with_unused_gas_weight(1).set(json!({
+            format!("{}.{}", community.handle, DEVHUB_COMMUNITY_FACTORY)
+            : {
+                "profile": {
+                    "name": community.name,
+                    "image": {
+                        "url": community.logo_url,
+                    }
+                }
+            }
+        }));
+    }
+
+    pub fn add_community_announcement(
+        &mut self,
+        handle: CommunityHandle,
+        announcement_post: String,
+    ) {
+        let _ = self
+            .get_editable_community(&handle)
+            .expect("Only community admins and hub moderators can create announcement");
+
+        require!(env::prepaid_gas() >= ADD_COMMUNITY_ANNOUNCEMENT_GAS, "Require at least 30 Tgas");
+        ext_social_db::ext(SOCIAL_DB.parse().unwrap()).with_unused_gas_weight(1).set(json!({
+            format!("{}.{}", handle, DEVHUB_COMMUNITY_FACTORY)
+            : {
+                "post": {
+                    "main": to_string(&json!({"type": "md", "text": announcement_post})).unwrap(),
```

Comment:
For more future flexibility and support for images, I think this function should take the whole post JSON, rather than just the text field.

"post": { "main": announcement_post }

Reply:
Sounds good